src/App.js

```diff
@@ -2,15 +2,29 @@ import "./App.css";
 import Header from "./components/Header";
 import Nav from "./components/Nav";
 import ArticlesList from "./components/ArticlesList";
-import { Route, Routes } from "react-router-dom";
+import SingleArticle from "./components/SingleArticle";
+import { Route, Routes, useNavigate } from "react-router-dom";
+import React, { useState } from "react";
 
 function App() {
+  const [showArticlesList, setShowArticlesList] = useState(false);
+  const navigate = useNavigate();
+
+  const handleHeaderClick = () => {
+    setShowArticlesList(!showArticlesList);
+    navigate("/articles");
+  };
```

Comment:
this `handleHeaderClick` isn't entirely necessary as you could wrap the `<h1>` tag inside of `Header` in a `Link` tag and link it to the relevant path `\`
this would make your code a bit cleaner :)